settings/factory.py

```diff
@@ -120,3 +124,16 @@ def get_writer(self):
       writer = RedisWriter(writer_conf, mapper, rank_field)
 
     return writer
+
+  def get_consumer_group_id(self, reader_conf):
+    try:
+      if self.config.get_consumer_group_mode() == "hostname":
+        # todo: need to log the exception
```

Comment:
you need to return the consumer group name from here right?

Reply:
yes, i will returning from here